rfcs/0000-User&HostIntegrationRFC.md

```diff
@@ -0,0 +1,254 @@
+asset.cost_center_hierarchy	| keyword |	Engineering	 | - | -                                | Additional cost center information associated with the asset (user or host).			
+asset.status	    | keyword         |	ACTIVE      | - | -                                 | Current status of the asset in the inventory datasource.			
+asset.last_status_change_date	| date |	June 5, 2023 @ 18:25:57.000	| - | -             | The most recent date/time when the asset.status was updated.			
+asset.create_date	            | date |	June 5, 2023 @ 18:25:57.001	| - | -             | For users, it's the hire date, for other assets, it's the in service date.			
+asset.end_date	                | date |	June 5, 2023 @ 18:25:57.002	| - | -             | For users, it's the termination date, for other assets, it's the out-of-service date.			
+asset.first_seen	            | date |	June 5, 2023 @ 18:25:57.003	| - | -             | The earliest date/time at which this asset was observed.  For assets added via config, this would remain empty until the asset was observed.			
+asset.last_seen	                | date |	June 5, 2023 @ 18:25:57.004	| - | -             | The most recent date/time at which this asset was observed.  Would remain empty until the asset was observed.			
+asset.last_updated	            | date |	June 5, 2023 @ 18:25:57.005	| - | -             | The most recent date/time this asset was updated in the inventory datasource.		
+asset.serial_number	            | keyword	| C02FG1G1MD6T	| - | -             |		Serial number of the asset.
+asset.tags	                    | keyword	  | watch, mdmaccess		| - | -             |	Tags assigned at the MDM.
+asset.assigned_users	          | keyword	  | user1@email.com, user2@email.com		| - | -             |	List of users assigned to the asset.
+asset.assigned_users_are_admin	| boolean	  | TRUE	| - | -             |		Flag to identify if the assigned users have admin priviledges.
+asset.is_managed	              | boolean	  | TRUE			| - | -             | If asset is managed by the organization.
+asset.last_enrolled_date	      | date	    | June 5, 2023 @ 18:25:57.005		| - | -             |	The most recent date/time the asset checked in with MDM.
+asset.data_classification	      | keyword	  | restricted		| - | -             |	Data classification tier for the asset.
+
+#### Nesting of existing risk.* fields under asset object
+* We have a set of risk.* fields in ECS which can be further nested under the asset.* object. Reference to [Risk RFC](https://github.com/elastic/ecs/blob/main/rfcs/text/0031-risk-fields.md).
+
+
+
+### Proposed New Fields for Host object
+Field | Type | Example | Description
+--- | --- | --- | ---
+host.os.build	| keyword		| 22F66   | Host OS Build information
+host.installed_extensions 	| keyword	  | Nested objects	  | List of installed extensions along with their metadata
```

Comment:
What is the motivation for adding `installed_extensions` and `installed_applications` underneath `host.*` instead of the proposed `asset.*` field set?

Reply:
I'm against about crowding the existing `host.*` field set if the main use for these new fields is tied to asset and entity store use cases. The `host.*` fields are mostly populated by the [add_host_metadata](https://www.elastic.co/guide/en/beats/filebeat/current/add-host-metadata.html) processor for each event, and we wouldn't want to add the list of `installed_extensions` and `installed_applications` to every produced event from the host.

In my own experience, while VM/container instances are tied to an underlying physical host, the instances are also treated as individual compute assets: unique owners, cost center, business owners, admins.

Reply:
> The `host.*` fields are mostly populated by the [add_host_metadata](https://www.elastic.co/guide/en/beats/filebeat/current/add-host-metadata.html) processor for each event, and we wouldn't want to add the list of `installed_extensions` and `installed_applications` to every produced event from the host.

The reference and elaboration are helpful in understanding the complexity. Let's move these two fields under assets.*.

Reply:
I addressed this as part of this commit - https://github.com/elastic/ecs/pull/2215/commits/fd289d1eac3c22a64e80d18989d64443ca8a6814